src/docs/app-monitoring/user-defined-monitoring.md

```diff
@@ -0,0 +1,221 @@
+
+Gauges are typically used for measured values like temperatures or current memory usage, but also "counts" that can go up and down, like the number of concurrent requests.​
+
+https://prometheus.io/docs/concepts/metric_types/#gauge
+
+### Histogram and Summaries
+
+**Histograms** and **summaries** both sample observations, typically request durations or response sizes. They track the number of observations and the sum of the observed values, allowing you to calculate the average of the observed values.​
+
+You can use both summaries and histograms to calculate so-called `φ-quantiles`, where 0 ≤ φ ≤ 1. The `φ-quantile` is the observation value that ranks at number `φ*N among the N observations`.
+
+Examples for φ-quantiles:
+
+- The `0.5-quantile` is known as the `median - 50th percentaile`.
+- The `0.95-quantile` is the `95th percentile`.​
+
+The essential difference between summaries and histograms is that summaries calculate streaming `φ-quantiles` on the client side and expose them directly, while histograms expose bucketed observation counts and the calculation of quantiles from the buckets of a histogram happens on the server side using the `histogram_quantile()` function.​
+
+https://prometheus.io/docs/practices/histograms/
+
+## Expose the metrics from your app
+
+Now that your app is generating a metrics endpoint, expose it with a Service.​ The key part is making sure the ports match up with what your library is using.
+
+```yaml
+apiVersion: v1​
+kind: Service​
+metadata:​
+  labels:​
+    app: myapp​
+  name: myapp​
+spec:​
+  ports:​
+  - name: metrics​
+    port: 8000​
+    protocol: TCP​
+    targetPort: 8000​
+  selector:​
+    app: myapp​
+  sessionAffinity: None​
+  type: ClusterIP 
+```
+
+## Create a ServiceMonitor
+
+To use the metrics exposed by your service, you must configure OCP monitoring to scrape metrics from the /metrics endpoint. You can do this using a ServiceMonitor custom resource definition (CRD) that specifies how a service should be monitored.
+
+```yaml
+apiVersion: monitoring.coreos.com/v1​
+kind: ServiceMonitor​
+metadata:​
+  labels:​
+    app: myapp​
+  name: myapp​
+spec:​
+  endpoints:​
+  - interval: 30s​
+    port: metrics​
+    scheme: http​
+  selector:​
+    matchLabels:​
+      app: myapp
+```
+
+![user defined monitoring](../../images/user-defined-monitoring.png)
+
+## Querying Prometheus
+
+PromQL is a rich language​ and there’s a lot you can do with PromQL to manipulate your metrics and make useful graphs. The docs are quite extensive.​
+
+- https://prometheus.io/docs/prometheus/latest/querying/basics/
+
+Note: Data is stored for 15 days currently.​
+
+## Sysdig Monitor
+
+You can now let Sysdig agent collecting your custom metrics and show them in the Sysdig console. Add `prometheus.io/scrape=true` annotation set in your pod. The Sysdig agent will scrape your application pod and send its `/metriics` to the Sysdig console.
```

Comment:
```suggestion
You can now let Sysdig agent collecting your custom metrics and show them in the Sysdig console. Add `prometheus.io/scrape=true` annotation set in your pod. The Sysdig agent will scrape your application pod and send its `/metrics` to the Sysdig console.
```

Reply:
btw here are some instruction on exposing app metrics to sysdig: https://docs.developer.gov.bc.ca/sysdig-monitor-set-up-advanced-functions/#use-service-discovery-to-import-application-metrics-endpoints

Reply:
Thanks! I will add that link.